doc/tutorials/setup_openstack_libvert.rst

```diff
@@ -0,0 +1,283 @@
+
+   ``make``
+
+   ``sudo make install``
+
+4. Edit and enable these options in ``/etc/libvirt/libvirtd.conf``:
+
+   ``unix_sock_ro_perms = "0777"``
+
+   ``unix_sock_rw_perms = "0777"``
+
+   ``unix_sock_admin_perms = "0777"``
+
+5. Run the following command:
+
+   ``sudo systemctl daemon-reload``
+
+
+Set up OpenStack
+****************
+
+Use DevStack to install OpenStack. Refer to the `DevStack instructions <https://docs.openstack.org/devstack/>`_.
+
+1. Use the latest maintenance branch **stable/train** to ensure OpenStack stability:
+
+   ``git clone https://opendev.org/openstack/devstack.git -b stable/train``
+
+2. Go to the devstack directory (cd devstack) and apply the following patch:
+
+   ``0001-devstack-installation-for-acrn.patch``
+
+3. Edit ``lib/nova_plugins/hypervisor-libvirt``:
+
+   Change ``xen_hvmloader_path`` to the location of your OVMF image file. A stock image is included in the ACRN source tree (devicemodel/bios/OVMF.fd).
+
+4. Copy the attached ``local.conf`` to ``devstack/``.
+
+.. Note::
+   Now is a great time to take a snapshot of the container using ``lxc snapsho``. If the OpenStack installation fails, manually rolling back to the previous state can be difficult. Currently, no step exists to reliably restart OpenStack after restarting the container.
+
+5. Install OpenStack:
+
+   ``execute ./stack.sh in devstack/``
+
+   The installation should take about 20-30 minutes. Upon successful installation, the installer reports the URL of OpenStack’s management interface. This URL is accessible from the native Ubuntu.
+
+   .. code-block:: none
+
+      …
+
+      Horizon is now available at http://10.117.165.131/dashboard
+
+      …
+
+      2020-04-09 01:21:37.504 | stack.sh completed in 1755 seconds.
+
+6. Verify in ``systemctl status libvirtd.service`` that libvirtd is active
+   and running.
+
+7. Set up SNAT for OpenStack instances to connect to the external network.
+
+   a. Inside the container, use ``ip a`` to identify the ``br-ex`` bridge
+      interface. ``br-ex`` should have two IPs. One should be visible to the native Ubuntu’s ``acrn-br0`` interface (e.g. inet 192.168.1.104/24). The other one is internal to OpenStack (e.g. inet 172.24.4.1/24). The latter corresponds to the public network in OpenStack.
+
+   b. Set up SNAT to establish a link between ``acrn-br0`` and OpenStack.
+      For example:
+
+      ``sudo iptables -t nat -A POSTROUTING -s 172.24.4.1/24 -o br-ex -j SNAT --to-source 192.168.1.104``
+
+Final Steps
+***********
+
+1. Create OpenStack instances.
+
+   - OpenStack logs to systemd journal
+   - libvirt logs to /var/log/libvirt/libvirtd.log
+
+   You can now use the URL to manage OpenStack in your native Ubuntu:
+     admin/intel123
+
+2. Create a router between **public** (external network) and **shared**
+   (internal network) using this as a reference:
+
+   https://www.itzgeek.com/how-tos/linux/centos-how-tos/configure-openstack-networking-enable-access-vm-instances.html#Create_Router
```

Comment:
@peterfang, @gvancuts mentioned that including a link to instructions from itzgeek.com (which is not Intel or Linux) might not be a good idea. Can you provide the actual steps that are needed here?

Reply:
I have updated this section,